graphql/src/execution/resolver.rs

```diff
@@ -95,11 +139,24 @@ pub trait Resolver: Clone + Send + Sync {
     // Resolves an abstract type into the specific type of an object.
     fn resolve_abstract_type<'a>(
         &self,
-        _schema: &'a s::Document,
+        schema: &'a s::Document,
         _abstract_type: &s::TypeDefinition,
-        _object_value: &q::Value,
+        object_value: &q::Value,
     ) -> Option<&'a s::ObjectType> {
-        None
+        let concrete_type_name = match object_value {
+            // All objects contain `__typename`
+            q::Value::Object(data) => match &data["__typename"] {
+                q::Value::String(name) => name.clone(),
+                _ => unreachable!("__typename must be a string"),
+            },
+            _ => unreachable!("abstract type value must be an object"),
```

Comment:
How sure are we that we're not going to hit this by accident? If not 100%, it would be useful to log things like the `__typename` value/type and the object value in these two exceptions.

Reply:
afaict only objects can implement interfaces in graphql. I'm being agressive with panics in query processing because they allow us to tell bugs from user errors. We can prevent these from crashing the node by differentating panics that occur during query processing.